File: common/src/main/java/com/ssblur/scriptor/registry/words/OffsetDescriptors.java
package com.ssblur.scriptor.registry.words;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.word.descriptor.target.OffsetDescriptor;

@SuppressWarnings("unused")
public class OffsetDescriptors {
  private static final WordRegistry INSTANCE = WordRegistry.INSTANCE;

  public final Descriptor MOVE_RIGHT = INSTANCE.register("move_right", new OffsetDescriptor().right());
  public final Descriptor COPY_RIGHT = INSTANCE.register("copy_right", new OffsetDescriptor().duplicate().right());
  public final Descriptor MOVE_LEFT = INSTANCE.register("move_left", new OffsetDescriptor().left());
  public final Descriptor COPY_LEFT = INSTANCE.register("copy_left", new OffsetDescriptor().duplicate().left());
  public final Descriptor MOVE_FORWARDS = INSTANCE.register("move_forwards", new OffsetDescriptor().forward());
  public final Descriptor COPY_FORWARDS = INSTANCE.register("copy_forwards", new OffsetDescriptor().duplicate().forward());
  public final Descriptor MOVE_BACKWARDS = INSTANCE.register("move_backwards", new OffsetDescriptor().backwards());
  public final Descriptor COPY_BACKWARDS = INSTANCE.register("copy_backwards", new OffsetDescriptor().duplicate().backwards());
  public final Descriptor MOVE_UP = INSTANCE.register("move_up", new OffsetDescriptor().up());
  public final Descriptor COPY_UP = INSTANCE.register("copy_up", new OffsetDescriptor().duplicate().up());
  public final Descriptor MOVE_DOWN = INSTANCE.register("move_down", new OffsetDescriptor().down());
  public final Descriptor COPY_DOWN = INSTANCE.register("copy_down", new OffsetDescriptor().duplicate().down());
}

File: common/src/main/kotlin/com/ssblur/scriptor/registry/words/OffsetDescriptors.kt
package com.ssblur.scriptor.registry.words

import com.ssblur.scriptor.registry.words.WordRegistry.register
import com.ssblur.scriptor.word.descriptor.target.OffsetDescriptor

@Suppress("unused")
object OffsetDescriptors {
  val MOVE_RIGHT = register("move_right", OffsetDescriptor(1.05).right())
  val COPY_RIGHT = register("copy_right", OffsetDescriptor(1.25).duplicate().right())
  val MOVE_LEFT = register("move_left", OffsetDescriptor(1.05).left())
  val COPY_LEFT = register("copy_left", OffsetDescriptor(1.25).duplicate().left())
  val MOVE_FORWARDS = register("move_forwards", OffsetDescriptor(1.05).forward())
  val COPY_FORWARDS = register("copy_forwards", OffsetDescriptor(1.25).duplicate().forward())
  val MOVE_BACKWARDS = register("move_backwards", OffsetDescriptor(1.05).backwards())
  val COPY_BACKWARDS = register("copy_backwards", OffsetDescriptor(1.25).duplicate().backwards())
  val MOVE_UP = register("move_up", OffsetDescriptor(1.05).up())
  val COPY_UP = register("copy_up", OffsetDescriptor(1.25).duplicate().up())
  val MOVE_DOWN = register("move_down", OffsetDescriptor(1.05).down())
  val COPY_DOWN = register("copy_down", OffsetDescriptor(1.25).duplicate().down())
}
